elastic/kibana
# [discover] do not show old results on error

Prerequisite for https://github.com/elastic/kibana/issues/167904

### Test
1. install sample web logs
2. Open discover
3. Select time range that displays results
4. Add filter
    ```
    {
      "error_query": {
        "indices": [
          {
            "error_type": "exception",
            "message": "local shard failure message 123",
            "name": "kibana_sample_data_logs"
          }
        ]
      }
    }
    ```
5. Verify discover should display `EmptyPrompt` showing error message instead of callout with stale data.
    <img width="500" alt="Screenshot 2023-11-07 at 9 23 43 AM" src="https://github.com/elastic/kibana/assets/373691/60237f8c-dcf5-4b4a-991f-4a9f073048af">
